### ISSUE TITLE
Confusion about dense model？

### ISSUE THREAD

**Issue by XGuider** [VITA-Group/GAN-Slimming#6]
Thank you best work,i meet some confusion about dense model.why need dense model?
The dataset is already one-to-one. 
I think we can skip directly to the fourth step without a dense model, 
but I don’t know if my idea is correct. Hope you can help me.

**Comment by XGuider** [comment]
if i use gs.py for other network,whether i need change model?

**Comment by htwang14** [comment]
Thanks for your interest in our work. GAN training is unstable due to the minimax optimization process, and combining it with model compression makes it even more unstable due to the use of non-differentiable sparse constraints. Using the dense model for distillation stabilizes the training process and empirically achieves better results. You can definitely directly compress GAN without the dense model, but it won't achieve as good a performance, as shown in our ablation study.

**Comment by htwang14** [comment]
> if i use gs.py for other network,whether i need change model?

Yes, you will need to replace CycleGAN (the model currently used) with your own network.
